test/tools/libtesteth/TestHelper.cpp

```
@@ -242,23 +244,69 @@ void replaceLLLinState(json_spirit::mObject& _o)
+std::vector<boost::filesystem::path> getFiles(boost::filesystem::path const& _dirPath, std::set<string> const _extentionMask, std::string const& _particularFile)
 {
-	vector<boost::filesystem::path> jsonFiles;
-	if (!_particularFile.empty())
+	vector<boost::filesystem::path> files;
+	for(auto const& ext: _extentionMask)
 	{
-		boost::filesystem::path file = _dirPath / (_particularFile + ".json");
-		if (boost::filesystem::exists(file))
-			jsonFiles.push_back(file);
+		if (!_particularFile.empty())
+		{
+			boost::filesystem::path file = _dirPath / (_particularFile + ext);
+			if (boost::filesystem::exists(file))
+				files.push_back(file);
+		}
+		else
+		{
+			using Bdit = boost::filesystem::directory_iterator;
+			for (Bdit it(_dirPath); it != Bdit(); ++it)
+			{
+				if (boost::filesystem::is_regular_file(it->path()) && it->path().extension() == ext)
+					files.push_back(it->path());
+			}
+		}
 	}
-	else
+	return files;
+}
+
+json_spirit::mValue convertYamlNodeToJson(YAML::Node _node)
+{
+	if (_node.IsNull())
+		return json_spirit::mObject();
+
+	if (_node.IsScalar())
 	{
-		using Bdit = boost::filesystem::directory_iterator;
-		for (Bdit it(_dirPath); it != Bdit(); ++it)
-			if (boost::filesystem::is_regular_file(it->path()) && it->path().extension() == ".json")
-					jsonFiles.push_back(it->path());
+		json_spirit::mValue value;
+		if (_node.Tag() == "tag:yaml.org,2002:int")
+			value = _node.as<int>();
+		else
+			value = _node.as<std::string>();
+		return value;
 	}
-	return jsonFiles;
+
+	if (_node.IsMap())
+	{
+		json_spirit::mObject jObject;
+		for (auto const& i : _node)
+			jObject[i.first.as<std::string>()] = convertYamlNodeToJson(i.second);
+		return jObject;
+	}
+
+	if (_node.IsSequence())
+	{
+		json_spirit::mArray jArray;
+		for (std::size_t i = 0; i < _node.size(); i++)
+			jArray.push_back(convertYamlNodeToJson(_node[i]));
+		return jArray;
+	}
+
+	return json_spirit::mValue();
```

I prefer an error here, rather than silently ignoring an unknown.

---

ok. will add boost error here